juju/juju
# Rename container spec to pod spec and add enhanced features

## Description of change

We enhance the spec construct used to allow units to configure k8s pods. Specifically we allow for the service to be omitted, and for there to be more than one container per pod. To reflect the nature of the change, the container-spec artefacts are renamed to pod-spec. The rename covers:
- hook command
- api facade methods
- state collection and data model 

Files are also renamed which blows out the diff as these appear as separate delete and add.

Once this lands, the caas base layer and caas charms will need to be updated and the charms rebuilt.

## QA steps

Deploy a k8s model.
Deploy mysql charm and ensure it works.
Modify mysql charm to omit service and deploy again, ensure it works.
